saleor/graphql/order/mutations/fulfillments.py

```diff
@@ -119,18 +141,57 @@ def clean_lines(cls, order_lines, quantities):
                     }
                 )
 
+    @classmethod
+    def check_warehouses_for_duplicates(cls, warehouse_ids):
+        for warehouse_ids_for_line in warehouse_ids:
+            duplicates = {
+                id
+                for id in warehouse_ids_for_line
+                if warehouse_ids_for_line.count(id) > 1
+            }
+            if duplicates:
+                raise ValidationError(
+                    {
+                        "warehouse": ValidationError(
+                            "Duplicated warehouse ID.", code=OrderErrorCode.UNIQUE
+                        )
+                    }
+                )
+
+    @classmethod
+    def check_lines_for_duplicates(cls, lines_ids):
+        duplicates = {id for id in lines_ids if lines_ids.count(id) > 1}
```

Comment:
There is also a new method for this `get_duplicated_values`:
https://github.com/mirumee/saleor/blob/master/saleor/graphql/core/utils/__init__.py#L115